force-app/main/default/classes/DMLOperations.cls

```diff
@@ -128,11 +134,22 @@ public with sharing class DMLOperations {
      * @param list of Opportunities to be changed then upserted.
      */
     public static void upsertOpportunityList(List<Opportunity> opportunities) {
-        for (Opportunity opp : opportunities) {
+          for(opportunity opp: opportunities){
             opp.StageName = 'Qualification';
             opp.CloseDate = Date.today().addMonths(3);
             opp.Amount = 50000;
-        }
+          }
+          //Updating an existing Opportunity
+          List<Opportunity> existingOpportunity = [SELECT Id, Name, StageName 
+          FROM Opportunity 
+          WHERE Name = 'Opp2'
+          Limit 1];
+          // If opportunity is greater zero run this condition
+          if(existingOpportunity.size() > 0){
+            Opportunity updateOpportunity = existingOpportunity[0];
+            updateOpportunity.stageName = 'Qualification';
+            opportunities.add(updateOpportunity);
+          }
```

Comment:
I'm not sure what the purpose is here of this code is. I don't think it is needed to solve the above issue?